src/Media/MediaManager.php

```diff
@@ -97,11 +103,21 @@ public function handleUploadImage(UploadImage $uploadImage): void
 
         $destinationIri = $this->iriGenerator->iri($destinationPath);
 
-        $this->filesystem->move(
+        // Move to the local file system media directory
+        $this->localFilesystem->copy(
             $uploadImage->getFilePath()->toNative(),
             $this->mediaDirectory . '/' . $destinationPath
         );
 
+        // Upload to the S3 bucket
+        $this->s3FileSystem->writeStream(
+            $destinationPath,
+            $this->localFilesystem->readStream($uploadImage->getFilePath()->toNative()),
+            [Config::OPTION_VISIBILITY => Visibility::PUBLIC]
+        );
+
+        $this->localFilesystem->delete($uploadImage->getFilePath()->toNative());
+
```

Comment:
I like that approach. Even in combination with a dedicated `ImageStorageProvider` where the local and S3 FlySystems can be instantiated instead of inside `bootstrap.php`